packages/protocol/contracts/libs/LibAddress.sol

```diff
@@ -43,4 +40,24 @@ library LibAddress {
             result = _result;
         } catch { }
     }
+
+    function isValidSignature(
+        address addr,
+        bytes32 hash,
+        bytes memory sig
+    )
+        internal
+        view
+        returns (bool valid)
+    {
+        if (
+            AddressUpgradeable.isContract(addr)
+                && IERC1271Upgradeable(addr).isValidSignature(hash, sig)
+                    == EIP1271_MAGICVALUE
```

Comment:
Currently this "merges" the IProver and IERC1271 together, right ?

Reply:
Kind of. It's less flexible than the previous one but it works with the tier-based prices better.